pvlib/mlfm.py

```diff
@@ -0,0 +1,803 @@
+'''
+This ``mlfm code`` module contains functions to analyse and predict
+performance of PV modules using the mechanistic performance (MPM) and
+loss factors models (LFM). The module also contains functions to display
+performance of PV modules using the mechanistic performance (MPM) and
+loss factors models (LFM)
+
+Authors : Steve Ransome (SRCL) and Juergen Sutterlueti (Gantner Instruments)
+Thanks to Cliff Hansen (Sandia National Laboratories)
+
+https://pvlib-python.readthedocs.io/en/stable/variables_style_rules.html#variables-style-rules
+
+https://github.com/python/peps/blob/master/pep-0008.txt
+'''
+
+import numpy as np
+import pandas as pd
+from scipy import optimize
+
+
+#  DEFINE REFERENCE MEASUREMENT CONDITIONS
+# or use existing definitions in pvlib
+
+# NAME  value  comment         unit     PV_LIB name
+#
+T_STC = 25.0  # STC temperature [C]      temperature_ref
+T_HTC = 75.0  # HTC temperature [C]
+G_STC = 1.0   # STC irradiance  [kW/m^2]
+G_LIC = 0.2   # LIC irradiance  [kW/m^2]
+
+
+#  Define standardised MLFM graph colours as a dict ``clr``
+
+clr = {
+    # parameter_clr colour            R   G   B
+    'irradiance':   'darkgreen',   # 000 064 000
+    'temp_module':  'red',         # 255 000 000
+    'temp_air':     'yellow',      # 245 245 220
+    'wind_speed':   'grey',        # 127 127 127
+
+    'i_sc':         'purple',      # 128 000 128
+    'r_sc':         'orange',      # 255 165 000
+    'i_ff':         'lightgreen',  # 144 238 144
+    'i_mp':         'green',       # 000 255 000
+    'i_v':          'black',       # 000 000 000 between i and v losses
+    'v_ff':         'cyan',        # 000 255 255
+    'v_mp':         'blue',        # 000 000 255
+    'r_oc':         'pink',        # 255 192 203
+    'v_oc':         'sienna',      # 160 082 045
+
+    'pr_dc':        'black',       # 000 000 000
+}
+
+
+def mlfm_meas_to_norm(dmeas, ref):
+    '''
+    Convert measured power, current and voltage to normalized values.
+
+    Parameters
+    ----------
+    dmeas : DataFrame
+        Measurements. Must include columns:
+
+        * `'poa_global_kwm2'` global plane of array irradiance [kW/m^2]
```

Comment:
I pointed out some suspected conflation in the code between using kW/m^2 units and using the (unitless) normalized irradiance G/(1000 W/m^2), where G is in W/m^2. I think it is important to avoid conflating this normalization with a units change.